networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/RateTracker.java

```diff
@@ -33,19 +33,19 @@ public RateTracker(
     this.timeProvider = timeProvider;
   }
 
-  // boundary: if a request comes in and remaining capacity is at least 1, then
-  // they can have the objects they request otherwise they get none.
-  public synchronized long wantToRequestObjects(final long objectCount) {
+  public synchronized long allowedRequestObjectsCount(final long objectCount) {
```

Comment:
nit: is there probably any better method name which emphasizes that object state is changed and requests are actually subtracted by this method? Kind of `take/poll/popRequests`  (the similar for `EthPeer` class)
Else the method invocation is pretending to be just a getter and the state mutation is lost inside `if` expression like: 
```java
    if (!peer.allowedToMakeRequest()
        || !peer.allowedToReceiveBlocks(
```
It would be quite easy to make a mistake in future updates IMO

Reply:
Right, the mutation isn't clear from the methods names. I renamed them to `popObjectRequests`, `popBlockRequests`, `popBlobSidecarRequests`, and `popRequest`